presto-main/src/main/java/com/facebook/presto/block/BlockUtils.java

```diff
@@ -55,4 +57,32 @@ else if (value instanceof String) {
             throw new IllegalArgumentException("Unsupported type: " + value.getClass().getName());
         }
     }
+
+    public static Object convertToType(Slice slice, Type type)
```

Comment:
There has to be a better way to do this. We shouldn't be adding (more) code to the engine that only works for certain types.